eclipse/src/saros/session/resources/validation/ResourceChangeValidator.java

```diff
@@ -0,0 +1,97 @@
+package saros.session.resources.validation;
+
+import org.apache.log4j.Logger;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.mapping.ModelProvider;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import saros.Saros;
+import saros.session.ISarosSession;
+import saros.session.User.Permission;
+
+/**
+ * This model provider is responsible for warning a session participant when trying to modify the
+ * resources of a shared project or the project itself in an unsupported way.
+ *
+ * <p>Ignoring the warnings will lead to inconsistencies. Note that we can't prevent these actions,
+ * just detect them and report them.
+ *
+ * <p>Currently detected:
+ *
+ * <ul>
+ *   <li>File and folder modifications as a user with {@link Permission#READONLY_ACCESS}.
+ *   <li>Deletion of a shared project.
+ *   <li>Renaming and moving of a shared project.
+ * </ul>
+ */
+/*
+ * FIXME there are several deltas that report changes which do not affect the
+ * file system. As this class did not work until Eclipse 4.x ??? nobody ever
+ * realized this
+ */
+public class ResourceChangeValidator extends ModelProvider {
+  private static final Logger log = Logger.getLogger(ResourceChangeValidator.class);
+
+  /** Error code for internal use, but we don't need it. */
+  private static final int ERROR_CODE = 0;
+
+  private static final IStatus MODIFYING_RESOURCES_ERROR_STATUS =
+      new Status(
```

Comment:
You could move this initialization to a static helper method to avoid code duplication as only the message ever changes.

Reply:
Done